rig/machine_control/machine_controller.py

```diff
@@ -1816,7 +1816,7 @@ def get_chip_info(self, x, y):
         info = self._send_scp(x, y, 0, SCPCommands.info, expected_args=3)
         num_cores = info.arg1 & 0x1F
         core_states = [consts.AppState(c)
-                       for c in struct.unpack("<18B", info.data)]
+                       for c in struct.unpack("<18B", info.data[:18])]
```

Comment:
Can you not do this with an unpack from? Rather than essentially repeating the "there'll be 18 bytes" thing.